connectors/es/sink.py

```diff
@@ -130,20 +129,16 @@ def _bulk_op(self, doc, operation=OP_INDEX):
 
     @tracer.start_as_current_span("_bulk API call", slow_log=1.0)
     async def _batch_bulk(self, operations, stats):
-        @retryable(retries=self.max_retires)
-        async def _bulk_api_call():
-            return await self.client.client.bulk(
-                operations=operations, pipeline=self.pipeline["name"]
-            )
-
         # TODO: treat result to retry errors like in async_streaming_bulk
         task_num = len(self.bulk_tasks)
 
         if self._logger.isEnabledFor(logging.DEBUG):
             self._logger.debug(
                 f"Task {task_num} - Sending a batch of {len(operations)} ops -- {get_mb_size(operations)}MiB"
             )
-        res = await _bulk_api_call()
+
+        # TODO: retry 429s for individual items here
```

Comment:
Could you please expand more on this comment? I don't fully understand how can we handle individual items errors.

Reply:
So now we have 500 items that we send, then if 1 fails we retry the whole batch.

What I wanna do in later PRs:

- See if error is retriable. If so, then put it back to the queue to re-ingest in a different batch
- If error is not retriable, raise following the resilience protocol we have (not raise if number of errors during ingestion does not exceed certain %, so if 0.5% of items are not ingested we don't raise, but if we're at 1% then we raise)
- Maybe have some sort of "dead letter queue" for these items? (that could happen cause of mappings blowing up with too many fields generated and such). This I probably won't do without extensive discussion and seeing if we even want that